tests/test_fisherz_test.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+from pywhy_stats.fisher_z_test import fisherz
+
+seed = 12345
+
+
+def test_fisher_z():
+    """Test Fisher Z test for Gaussian data."""
+
+    # We construct a SCM where X1 -> Y <- X and Y -> Z
+    # so X1 is independent from X, but conditionally dependent
+    # given Y or Z
+    rng = np.random.default_rng(seed)
```

Comment:
Avoid setting a random seed. Instead use the flaky decorator (e.g., see here https://github.com/py-why/dowhy/blob/main/tests/gcm/independence_test/test_kernel.py#L10). It would be very unlikely if this fails more than 3x in a row (then there is probably an issue). If you set a random seed, then the following tests will also be affected (if you don't restore the random state) and you ran into the issue that you might "optimize" for this particular seed.

In this regard, make each test also as a separate test. Here, again, consider the "given-when-then"-pattern (also see https://github.com/py-why/dowhy/blob/main/tests/gcm/independence_test/test_kernel.py#L11 as example).

Reply:
The random seed here is not set, instead we use a deterministic random generator (new in numpy) that doesn't affect outside the function.

Re flaky: The issue I have w/ flaky in this case is if we use it all over the place is that there is still the chance of failure in the test right? So in principle, out of 100 CI runs, we may have to re-run the whole CI pipeline a few times at least, which is unscalable eventually, and also make the CIs slower cuz now your # tests * 3-5.

I wonder if we can instead create a fixture of random seeds that are inherited by each test, such that our tests are each deterministic wrt 2-3 random states and low likelihood that we can overfit to a seeded random state?

Reply:
This issue is neglectable. We have/had ~200+ unit tests in DoWhy where the majority uses flaky and if a test is constant failing then its mostly because there is an actual issue or the test requirements were too strict. That's also why it is particularly important to separate these tests into multiple ones, so the likelihood of random failing is low. There is of course some runtime impact, but then again, we would expect the failure to be the exception and, thus, neglectable.

If we ever run into the actual problem of constant failing, then we can look into alternatives. But I wouldn't expect this to be an issue when we don't have at least a few thousand tests.

Reply:
What about we try it out with flaky. If it really becomes an issue, we can look into alternative ways.